### DISCUSSION TITLE
Wahoo Elemnt Bolt v2 does not configure

### DISCUSSION THREAD

**Discussion by jstevensog** [NightscoutFoundation/xDrip#2337] [Q&A]
While I believe this is an issue, I have been instructed to open a discussion instead.

Attempting to set up a Wahoo Elemnt Bolt v2 with USB C. Have an OTG adaptor, and if I plug the Bolt v2 into my phone I can browse it fine. However, using the "start usb configuration" command, it does not find the Bolt v2, although the phone finds it and wants to open it. Any advice appreciated.

**Comment by Navid200** [comment]
Thanks.

Have you seen this?
https://github.com/NightscoutFoundation/xDrip/issues/1654

**Comment by jstevensog** [comment]
Yes, but it is a differente problem.  The new V2 Elemnt Bolt has a USB C
connector, and all attempts to have it act as a direct collector for my G6
fails at the first hurdle.
While it is definitely detected and mounted as storage to my phone, the
xDrip "start usb configuration" does not detect it to configure it.
This is likely due to some setting on the phone's USB, but I am having
difficulty finding it.  Android 13/Pixel 5.
Cheers

On Wed, 14 Sept 2022 at 13:55, Navid ***@***.***> wrote:

> Thanks.
>
> Have you seen this?
> #1654 <https://github.com/NightscoutFoundation/xDrip/issues/1654>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NightscoutFoundation/xDrip/discussions/2337#discussioncomment-3640784>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACCCZ45RS2IQMNZ7GVAMFPTV6FEBVANCNFSM6AAAAAAQL77ETQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***
> com>
>


-- 
John Stevens
"You are how you live, not what you have."

**Reply by Navid200** [reply]
If you find a solution, obviously, you can open a PR as you have been here before me.  And thanks for your service.

If not, let's see if anyone has any hope that this is doable or not.  If yes, we can make this an issue.  So, we can reopen your issue.

Just to let you know, I pushed for the change you see.  I have insisted that people open discussions instead of issues no matter what.  If someone opens an issue and says that well I am sure this is an issue, how can we expect everyone else not to do the same?

There was a time we were approaching a total number of open issues of 900!  There was absolutely no way to know what was going on and what issue was falling behind and what issue was critical and what issue was good to have.  There were just too many of them.
The reason we now only have 150 open issues is the time we have spent cleaning them up and insisted that people open discussions.
I don't want to go back to 900 open issues.

Many of the discussions conclude by just helping the user find out how to use xDrip as is.  
There are even discussions that directly result in an edit to the code without needing an issue.
Some discussions result in opening an issue, but not all.

I have insisted to change our approach to this.  I am sorry if it is inconvenient.  But, the alternative is really ugly.
Thanks for your patience.

**Comment by jstevensog** [comment]
No issue with the change at all.  I feel it is the right approach.  The
only issue is that since I had been out of the loop for some time (work
getting in the way of interesting things) I was not aware.
Apologies if I came across a little snarky or miffed, that was not really
the case.
I am fairly sure there is a USB setting somewhere that is preventing the
xDrip configurator getting to the Bolt v2.  Something like a default
protocol (File Transfer, rather than Data).
I will poke around and if I find the problem will let everyone know how to
solve it.
Cheers

On Thu, 15 Sept 2022 at 10:42, Navid ***@***.***> wrote:

> If you find a solution, obviously, you can open a PR as you have been here
> before me. And thanks for your service.
>
> If not, let's see if anyone has any hope that this is doable or not. If
> yes, we can make this an issue. So, we can reopen your issue.
>
> Just to let you know, I pushed for the change you see. I have insisted
> that people open discussions instead of issues no matter what. If someone
> opens an issue and says that well I am sure this is an issue, how can we
> expect everyone else not to do the same?
>
> There was a time we were approaching a total number of open issues of 900!
> There was absolutely no way to know what was going on and what issue was
> falling behind and what issue was critical and what issue was good to have.
> There were just too many of them.
> The reason we now only have 150 open issues is the time we have spent
> cleaning them up and insisted that people open discussions.
> I don't want to go back to 900 open issues.
>
> Many of the discussions conclude by just helping the user find out how to
> use xDrip as is.
> There are even discussions that directly result in an edit to the code
> without needing an issue.
> Some discussions result in opening an issue, but not all.
>
> I have insisted to change our approach to this. I am sorry if it is
> inconvenient. But, the alternative is really ugly.
> Thanks for your patience.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NightscoutFoundation/xDrip/discussions/2337#discussioncomment-3649265>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACCCZ4Y2EL6J2BE2GTB3XWTV6JWGBANCNFSM6AAAAAAQL77ETQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***
> com>
>


-- 
John Stevens
"You are how you live, not what you have."

**Comment by jstevensog** [comment]
Hi All,
So, I decided to try a specifically marked as OTG cable.  Result is the
same.  xDrip cannot connect to the Bolt v2 over USB-C in order to configure
it.  It always tries to open the file browser, so it is detecting it.
I cannot get the phone to do anything else.  The connection does not allow
changing the connection type from file transfer to anything else.
It may simply be that the Bolt v2 does not allow the previous usb
configuration to display BGL.
Does anyone have a contact at Wahoo that I could approach regarding this?
As it is not a "standard feature" of the Bolt v2, I doubt raising an issue
with their support team will help.
Cheers

On Thu, 15 Sept 2022 at 10:47, John Stevens ***@***.***> wrote:

> No issue with the change at all.  I feel it is the right approach.  The
> only issue is that since I had been out of the loop for some time (work
> getting in the way of interesting things) I was not aware.
> Apologies if I came across a little snarky or miffed, that was not really
> the case.
> I am fairly sure there is a USB setting somewhere that is preventing the
> xDrip configurator getting to the Bolt v2.  Something like a default
> protocol (File Transfer, rather than Data).
> I will poke around and if I find the problem will let everyone know how to
> solve it.
> Cheers
>
> On Thu, 15 Sept 2022 at 10:42, Navid ***@***.***> wrote:
>
>> If you find a solution, obviously, you can open a PR as you have been
>> here before me. And thanks for your service.
>>
>> If not, let's see if anyone has any hope that this is doable or not. If
>> yes, we can make this an issue. So, we can reopen your issue.
>>
>> Just to let you know, I pushed for the change you see. I have insisted
>> that people open discussions instead of issues no matter what. If someone
>> opens an issue and says that well I am sure this is an issue, how can we
>> expect everyone else not to do the same?
>>
>> There was a time we were approaching a total number of open issues of
>> 900! There was absolutely no way to know what was going on and what issue
>> was falling behind and what issue was critical and what issue was good to
>> have. There were just too many of them.
>> The reason we now only have 150 open issues is the time we have spent
>> cleaning them up and insisted that people open discussions.
>> I don't want to go back to 900 open issues.
>>
>> Many of the discussions conclude by just helping the user find out how to
>> use xDrip as is.
>> There are even discussions that directly result in an edit to the code
>> without needing an issue.
>> Some discussions result in opening an issue, but not all.
>>
>> I have insisted to change our approach to this. I am sorry if it is
>> inconvenient. But, the alternative is really ugly.
>> Thanks for your patience.
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/NightscoutFoundation/xDrip/discussions/2337#discussioncomment-3649265>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/ACCCZ4Y2EL6J2BE2GTB3XWTV6JWGBANCNFSM6AAAAAAQL77ETQ>
>> .
>> You are receiving this because you authored the thread.Message ID:
>> ***@***.***
>> .com>
>>
>
>
> --
> John Stevens
> "You are how you live, not what you have."
>


-- 
John Stevens
"You are how you live, not what you have."

**Comment by jamorham** [comment]
I will see what I can do for this.

**Reply by Navid200** [reply]
I have reopened the issue (https://github.com/NightscoutFoundation/xDrip/issues/2335) and assigned to you.  Please undo or let me know to undo any of that.

**Reply by bobcosine** [reply]
I see two issues as of Feb 2024.  One is connecting xdrip+ to Wahoo Bolt v2 using USB C.  Any progress on that?
If xdrip+ can display BG on Wahoo Bolt v2 then both G6 and  G7 can display unless I am missing something.  I would be happy to work on this but want to know if any progress has been made since Sept 2022.  Thanks.

**Reply by Navid200** [reply]
I am not working on this.  I cannot speak for anyone else.

**Comment by jstevensog** [comment]
Hi All,
Any update on this?
Cheers

On Tue, 20 Sept 2022 at 20:48, JamOrHam ***@***.***> wrote:

> I will see what I can do for this.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NightscoutFoundation/xDrip/discussions/2337#discussioncomment-3689169>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACCCZ44YZEMUCQRBNVIVXOTV7GI7RANCNFSM6AAAAAAQL77ETQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***
> com>
>


-- 
John Stevens
"You are how you live, not what you have."

**Comment by atomspring** [comment]
Very interested in this as well. 

What steps were taking place under the hood for the bolt v1? Maybe one of the steps is no longer directly supported and needs a workaround.

**Comment by nvanderperren** [comment]
I have the same issue with the elemnt roam v2

**Comment by jstevensog** [comment]
I would be happy to work on it, but don't really know who at Wahoo to work
with to get it analysed and working.
Cheers

On Wed, 14 Feb 2024 at 07:46, Navid ***@***.***> wrote:

> I am not working on this. I cannot speak for anyone else.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NightscoutFoundation/xDrip/discussions/2337#discussioncomment-8458537>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACCCZ4ZZ6AHCMBAND3MVWOLYTPGKXAVCNFSM6AAAAAAQL77ETSVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4DINJYGUZTO>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***
> com>
>


-- 
John Stevens
"You are how you live, not what you have."

**Reply by bobcosine** [reply]
Hey.  I do not know anyone at Wahoo but I will try to make a contact.  When it worked with the V1 Bolt it was @jamorham who wrote the code or at least @jamorham was the one to write the "how to" information.  The change as far as I can tell is to USB C and also whatever different code resides in the V2.  Still think it is Android based tho.

**Reply by CodyBradford** [reply]
I can confirm that the Element Bolt V2 with USB-C won't connect as per previous 'how to'. I've tried various devices and cables. I borrowed a V1 to test (micro USB) and it works fine
